packages/driver/cypress/integration/issues/3871_spec.js

```diff
@@ -0,0 +1,58 @@
+// https://github.com/cypress-io/cypress/issues/3871
+describe(`issue 3871 - Cypress invoke`, () => {
+  it(`should work on not nested properties`, () => {
+    cy.window().then((win) => {
+      win.val = 40
+      win.get = function () {
+        return this.val
+      }
+
+      cy.window().invoke(`get`).should(`eq`, 40)
+    })
+  })
+
+  it(`should work on 1 level nested properties`, () => {
+    cy.window().then((win) => {
+      win.obj = {
+        val: 41,
+        get () {
+          return this.val
+        },
+      }
+
+      cy.window().invoke(`obj.get`).should(`eq`, 41)
+    })
+  })
+
+  it(`should work on 2 level nested properties`, () => {
+    cy.window().then((win) => {
+      win.obj = {
+        innerObj: {
+          val: 42,
+          get () {
+            return this.val
+          },
+        },
+      }
+
+      cy.window().invoke(`obj.innerObj.get`).should(`eq`, 42)
+    })
+  })
+
+  it(`should work on 3 level nested properties`, () => {
```

Comment:
I went up to 3, this should work on any arbitrary depth, please let me know with testing up to level 3 suffices :slightly_smiling_face: